src/stirling/source_connectors/socket_tracer/protocols/mongodb/stitcher.cc

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "src/stirling/source_connectors/socket_tracer/protocols/mongodb/stitcher.h"
+
+#include <string>
+#include <utility>
+#include <variant>
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/strings/str_replace.h>
+#include "src/common/base/base.h"
+
+#include "src/stirling/source_connectors/socket_tracer/protocols/common/interface.h"
+#include "src/stirling/source_connectors/socket_tracer/protocols/mongodb/types.h"
+#include "src/stirling/utils/binary_decoder.h"
+
+namespace px {
+namespace stirling {
+namespace protocols {
+namespace mongodb {
+
+void FindMoreToComeResponses(
+    absl::flat_hash_map<mongodb::stream_id_t, std::deque<mongodb::Frame>>* resps, int* error_count,
+    mongodb::Frame* resp_frame, uint64_t* latest_resp_ts) {
+  // In a more to come message, the response frame's responseTo will be the requestID of the prior
+  // response frame.
+
+  // Find and insert all of the more to come frame(s) section data to the head response frame.
+  auto curr_resp = resp_frame;
+
+  while (curr_resp->more_to_come) {
+    // Find the next response's deque.
+    auto next_resp_deque_it = resps->find(curr_resp->request_id);
+    if (next_resp_deque_it == resps->end()) {
+      VLOG(1) << absl::Substitute(
+          "Did not find a response deque extending the prior more to come response. "
+          "requestID: $0",
+          curr_resp->request_id);
+      (*error_count)++;
+      break;
+    }
+
+    // Response deque containing the next more to come response frame.
+    auto& next_resp_deque = next_resp_deque_it->second;
+
+    // Find the next response frame from the deque with a timestamp just greater than the
+    // current response frame's timestamp.
+    auto next_resp_it = std::upper_bound(
+        next_resp_deque.begin(), next_resp_deque.end(), *latest_resp_ts,
+        [](const uint64_t ts, const mongodb::Frame& frame) { return ts < frame.timestamp_ns; });
+    if (next_resp_it->timestamp_ns < *latest_resp_ts) {
+      VLOG(1) << absl::Substitute(
+          "Did not find a response extending the prior more to come response. RequestID: $0",
+          curr_resp->request_id);
+      (*error_count)++;
+      break;
```

Comment:
Since this is inside of a function now, I think it would be better to return rather than break.
```suggestion
      return;
```